src/Routers/FunctionsRouter.js

```diff
@@ -56,18 +56,21 @@ export class FunctionsRouter extends PromiseRouter {
       log: req.config.loggerController,
       headers: req.config.headers,
       ip: req.config.ip,
-      jobName
-    };
-    const status = {
-      success: jobHandler.setSucceeded.bind(jobHandler),
-      error: jobHandler.setFailed.bind(jobHandler),
+      jobName,
       message: jobHandler.setMessage.bind(jobHandler)
-    }
+    };
+
     return jobHandler.setRunning(jobName, params).then((jobStatus) => {
       request.jobId = jobStatus.objectId
       // run the function async
       process.nextTick(() => {
-        jobFunction(request, status);
+        Promise.resolve().then(() => {
```

Comment:
why the extra promise?  Can't you just  
```
jobFunction(request)
  .then(jobHandler.setSucceeded)
  .catch(jobHandler.setFailed);
```
and it'll be exactly the same and clearer?

Reply:
jobFuncirton isn't guaranteed to return a promise object. can be anything.

Reply:
I see.  no likey, but i see.  so fine by me, but i'll be on the lookout for a more concise idiom ;)

Reply:
the alternative is:

```
let result;
try {
   result = jobFunction(request);
} catch(e) {
   result = Promise.reject(e);
}
if (!result instanceof Promise) {
   result = Promise.resolve(result);
}
result.then(() => {

}).catch(() => {
});
```